pkg/controller/statefulset/stateful_set.go

```diff
@@ -27,62 +26,51 @@ import (
 	clientv1 "k8s.io/client-go/pkg/api/v1"
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/tools/record"
+
 	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/api/v1"
 	apps "k8s.io/kubernetes/pkg/apis/apps/v1beta1"
 	"k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
 	"k8s.io/kubernetes/pkg/client/legacylisters"
 
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/util/errors"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/apimachinery/pkg/watch"
 	"k8s.io/client-go/util/workqueue"
 	"k8s.io/kubernetes/pkg/controller"
 
+	"fmt"
 	"github.com/golang/glog"
 )
 
 const (
 	// Time to sleep before polling to see if the pod cache has synced.
 	PodStoreSyncedPollPeriod = 100 * time.Millisecond
-	// number of retries for a status update.
-	statusUpdateRetries = 2
 	// period to relist statefulsets and verify pets
 	statefulSetResyncPeriod = 30 * time.Second
 )
 
 // StatefulSetController controls statefulsets.
 type StatefulSetController struct {
+	// client interface
 	kubeClient clientset.Interface
-
 	// newSyncer returns an interface capable of syncing a single pet.
 	// Abstracted out for testing.
-	newSyncer func(*pcb) *petSyncer
-
+	control StatefulSetControlInterface
```

Comment:
The reason behind the "additional complexity" is that as a future maintainer when I come across a client wrapper, it's really hard to reason about it w/o looking at the actual code. Bare clients are much easier to maintain. Not something that you need to remove from your code now but something to think about as we move forward.